### ISSUE TITLE
SaberFactory spams Critical errors in log while playing.

### ISSUE THREAD

**Issue by SteffoSpieler** [ToniMacaroni/SaberFactory#33]
As statet in the Title SaberFactory (or something else) is spamming Critical Errors into the log.

```
[CRITICAL @ 23:28:41 | UnityEngine] TrailElementCollection.MoveTailToHead () (at <95802429957344f68e874c5b2a6a4dc4>:0)
[CRITICAL @ 23:28:41 | UnityEngine] SaberFactory.Instances.Trail.SFTrail.LateUpdate () (at <74beae87560e412da13b14e6e1edf113>:0)
[CRITICAL @ 23:28:41 | UnityEngine] DivideByZeroException: Attempted to divide by zero.
```

If you need more info just ask.

**Comment by ToniMacaroni** [comment]
Can you try the newest update?
If this doesn't fix the issue please provide the saber you used.

**Comment by SteffoSpieler** [comment]
Still these ciritical logs.
I am using [these](https://modelsaber.com/Sabers/?id=1587312770&pc) Sabers.
(Also I am using vrmode=oculus)

**Comment by SteffoSpieler** [comment]
Well.. the old error is gone - but a new one gets spammed in the console:
```
[CRITICAL @ 16:31:47 | UnityEngine] ArgumentOutOfRangeException: Index was out of range. Must be non-negative and less than the size of the collection.
[CRITICAL @ 16:31:47 | UnityEngine] Parameter name: index
[CRITICAL @ 16:31:47 | UnityEngine] System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) (at <9577ac7a62ef43179789031239ba8798>:0)
[CRITICAL @ 16:31:47 | UnityEngine] System.ThrowHelper.ThrowArgumentOutOfRangeException () (at <9577ac7a62ef43179789031239ba8798>:0)
[CRITICAL @ 16:31:47 | UnityEngine] SaberFactory.Instances.Trail.AltTrail.RefreshSpline () (at <9b9d79bb2e7546c9bb69b5738d175460>:0)
[CRITICAL @ 16:31:47 | UnityEngine] SaberFactory.Instances.Trail.AltTrail.LateUpdate () (at <9b9d79bb2e7546c9bb69b5738d175460>:0)
```

I hope that helps - I'm still using the same sabers I linked in my other comment.

**Comment by ToniMacaroni** [comment]
I fixed it but the version isn't released yet

**Comment by ToniMacaroni** [comment]
Fixed in #49

**Comment by DevilPF** [comment]
Hi, it's still flooding the logs with 
```
[CRITICAL @ 13:46:25 | UnityEngine] System.ThrowHelper.ThrowArgumentOutOfRangeException (System.ExceptionArgument argument, System.ExceptionResource resource) (at <9577ac7a62ef43179789031239ba8798>:0)
[CRITICAL @ 13:46:25 | UnityEngine] System.ThrowHelper.ThrowArgumentOutOfRangeException () (at <9577ac7a62ef43179789031239ba8798>:0)
[CRITICAL @ 13:46:25 | UnityEngine] SaberFactory.Instances.Trail.AltTrail.RefreshSpline () (at <a51bb068eea949e8b42836459adf0422>:0)
[CRITICAL @ 13:46:25 | UnityEngine] SaberFactory.Instances.Trail.AltTrail.LateUpdate () (at <a51bb068eea949e8b42836459adf0422>:0)
```
even on 2.3.3

I'm using this sabers: https://modelsaber.com/Sabers/?id=1609570100&pc
